src/utils/apiRequests/apiClient.ts

```diff
@@ -2,57 +2,119 @@ import { APIError } from '@planet-sdk/common';
 import { getQueryString } from './getQueryString';
 
 export interface RequestOptions {
+  /**
+   * The HTTP method to be used for the request.
+   * Restricted to standard CRUD operations.
+   * @example 'GET' | 'POST' | 'PUT' | 'DELETE'
+   */
   method: 'GET' | 'POST' | 'PUT' | 'DELETE';
+  /**
+   * The URL or path for the API endpoint.
+   * Can be absolute (starting with http:// or https://)
+   * or relative (will be prepended with apiBaseUrl).
+   * @example '/users' or 'https://api.example.com/users'
+   */
   url: string;
-  data?: any;
+  /**
+   * Optional request payload for POST and PUT methods.
+   * Represents key-value pairs of data to be sent.
+   * @example { username: 'john_doe', email: 'john@example.com' }
+   */
+  data?: Record<string, unknown>;
+  /**
```

Comment:
Are POST and PUT requests without a payload possible? If not, perhaps the type should be updated to 

```
export type RequestOptions = {
  url: string;
  queryParams?: { [key: string]: string };
  additionalHeaders?: { [key: string]: string };
  authRequired?: boolean;
} & (
  | { method: 'GET' | 'DELETE'; data?: Record<string, unknown> }
  | { method: 'POST' | 'PUT'; data: Record<string, unknown> }
);
```

so that warnings are shown when a POST/PUT request does not include data.